### PR TITLE
Refactor calendars, introduce calendar customization

### PR DESCRIPTION
**Pull request by arkid15r**
## Proposed change
Prettify the code structure around recently updated calendars:
  - Split calendars into multiple files.
  - Add calendar customization mechanism (Islamic).
  - Remove `_islamic_to_gre` as unused.
  - Update tests: fix estimated holidays names.
  - Update Pakistan Ashura dates.

## Type of change

- [ ] New country/market holidays support (thank you!)
- [ ] Supported country/market holidays update (calendar discrepancy fix, localization)
- [x] Existing code/documentation/test/process quality improvement (best practice, cleanup, refactoring, optimization)
- [ ] Dependency upgrade (version update)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (a code change causing existing functionality to break)
- [ ] New feature (new python-holidays functionality in general)

## Checklist

- [x] I've followed the [contributing guidelines][contributing-guidelines]
- [x] This PR is filed against `beta` branch of the repository
- [x] This PR doesn't contain any merge conflicts and has clean commit history
- [x] The code style looks good: `make pre-commit`
- [x] All tests pass locally: `make test`, `make tox` (we strongly encourage adding tests to your code)
- [x] The related [documentation][docs] has been added/updated (check off the box for free if no updates is required)

[contributing-guidelines]: https://github.com/dr-prodigy/python-holidays/blob/beta/CONTRIBUTING.rst
[docs]: https://github.com/dr-prodigy/python-holidays/tree/beta/docs/source

### CODE REVIEW

File: holidays/countries/pakistan.py
Comment (arkid15r) [inline holidays/countries/pakistan.py:1]: @KJhellico, could you look closer on the Pakistan holidays changes (Ashura). I'm not sure if I've done everything right.

**Comment by KJhellico** [comment]
`exact_dates` realization: 🚀

File: holidays/countries/pakistan.py [hunk 1]
```diff
@@ -196,3 +76,94 @@ class PK(Pakistan):
...
+        2013: ((NOV, 13),),
+        2014: ((NOV, 3),),
+        2015: ((OCT, 23),),
+        2016: ((OCT, 11),),
+        2017: ((SEP, 30),),
+        2018: ((SEP, 21),),
+        2019: ((SEP, 9),),
+        2020: ((AUG, 29),),
+        2021: ((AUG, 18),),
+        2022: ((AUG, 8),),
```
Comment (KJhellico) [inline holidays/countries/pakistan.py:102]:
```suggestion
        2022: ((AUG, 9),),
```

File: tests/countries/test_pakistan.py [hunk 2]
```diff
@@ -150,9 +150,48 @@ def test_ashura(self):
...
+            ("2022-03-23", "Pakistan Day"),
+            ("2022-05-01", "Labour Day"),
+            ("2022-05-03", "Eid-ul-Fitr"),
+            ("2022-05-04", "Eid-ul-Fitr"),
+            ("2022-05-05", "Eid-ul-Fitr"),
+            ("2022-07-10", "Eid-ul-Adha"),
+            ("2022-07-11", "Eid-ul-Adha"),
+            ("2022-07-12", "Eid-ul-Adha"),
+            ("2022-08-07", "Ashura"),
+            ("2022-08-08", "Ashura"),
```
Comment (KJhellico) [inline tests/countries/test_pakistan.py:192]:
```suggestion
            ("2022-08-08", "Ashura"),
            ("2022-08-09", "Ashura"),
```

**Review by KJhellico — CHANGES_REQUESTED** [review]
It seems that in 2022, Ashura in Pakistan was on [August 8-9](https://www.geo.tv/latest/431501-muharram-2022-govt-announces-two-day-holiday-for-ashura).

**Comment by KJhellico** [comment]
And `BEGIN_RAMADAN_DATES` is missing.

File: holidays/countries/united_arab_emirates.py [hunk 3]
```diff
@@ -45,108 +43,55 @@ class UnitedArabEmirates(HolidayBase, InternationalHolidays, IslamicHolidays):
...
-            if year in dates_obs:
-                for date_obs in dates_obs[year]:
-                    self._add_holiday(name, *date_obs)
-            else:
-                self._add_mawlid_day(f"{name}* (*estimated)")
+        # Eid al-Fitr.
+        eid_al_adha = "Eid al-Fitr"
+        self._add_eid_al_fitr_day(eid_al_adha)
+        self._add_eid_al_fitr_day_two(f"{eid_al_adha} Holiday")
+        self._add_eid_al_fitr_day_three(f"{eid_al_adha} Holiday")
```
[the remark this answers is not in the document]
Comment (arkid15r) [reply holidays/countries/united_arab_emirates.py]: Renamed

**Comment by arkid15r** [comment]
> And `BEGIN_RAMADAN_DATES` is missing.

It was defined but unused so I cleaned that up.
It's been restored (I changed the name a bit though).

File: holidays/holiday_groups.py [hunk 4]
```diff
@@ -804,27 +842,47 @@ def _add_eid_al_fitr_day_four(self, holiday_name) -> Set[date]:
...
-                dt = self._add_holiday(holiday_name, dt)
-                if dt:
-                    dates.add(dt)
-        return dates
+
+        Appends customizable estimation label at the end of holiday name if
+        holiday date is an estimation.
+        """
+        added_dates = set()
+        estimated_label = getattr(self, "estimated", "estimated")
```
Comment (KJhellico) [inline holidays/holiday_groups.py:874]: In Pakistan (only use for now), this attribute is named `estimated_label`, need to synchronize :)

**Review by KJhellico — COMMENTED** [review]
LGTM.

**Review by KJhellico — APPROVED** [review]
Let's go!

**Pull request merged**